### Calc Submission Rank
https://www.kaggle.com/c/home-credit-default-risk

Show what my submission rank would have been for each model.

In [1]:
import pandas as pd
import numpy as np

In [4]:
leaderboard_df.columns

Index(['TeamId', 'TeamName', 'SubmissionDate', 'Score'], dtype='object')

In [44]:
leaderboard_df = pd.read_csv("data/home-credit-default-risk-publicleaderboard.csv")
leaderboard_df = leaderboard_df.sort_values('Score', ascending=True)
scores = leaderboard_df['Score'].values
# scores = np.sort(scores)
print("top",scores[:10]) 
print("bottom", scores[-10:])

top [0.25259 0.25365 0.25473 0.2549  0.25511 0.2553  0.25559 0.25576 0.25591
 0.25707]
bottom [0.81471 0.81502 0.81515 0.81562 0.81566 0.81584 0.81629 0.81652 0.81706
 0.81724]


In [7]:
my_submissions = {
    'dnn__weighted_cross_entropy_with_logits_exp/main_plus_all/submissions.csv': 0.50995,
    "xgboost_kaggle_higgs_exp/main_plus_all/submissions.csv": 0.68446,
    "xgboost_exp.max_depth_12/main_plus_all/submissions.csv": 0.50495,
    "xgboost_exp/main_plus_all_numeric_files/submissions.csv": 0.50044,
    "xgboost_exp/main_plus_all/submissions.csv": 0.50105,
    "random_forest_exp/main_plus_all_numeric_files/submissions.csv": 0.62165,
    "random_forest_exp/main_plus_all features": 0.62130,
    "dnn__default_exp/main_plus_all_numeric_files/submissions.csv": 0.50072,
    "dnn__default_exp/main_plus_all__softmax/submissions.csv": 0.25845,
    "dnn__default_exp/main_plus_all__softmax/submissions.csv": 0.50255
}

In [69]:
total_submissions = scores.shape[0]
print("top 10 submission scores",scores[::-1][:10]) 
print("bottom 10 submission scores", scores[::-1][-10:])

for k, v in my_submissions.items():
    print('\n'+"************")
    print(k,v)
    score_rank = np.searchsorted(scores, v, side="right")
    print("Ranked", total_submissions - score_rank, "of", total_submissions, "submissions")
    #
    print("My submission was worse than {:0.2f}% of submissions".format(
        float(np.sum(scores >= v)) / float(total_submissions)*100))
    print("My submission was better than {:0.2f}% of submissions".format(
        float(np.sum(scores < v)) / float(total_submissions)*100))
    print("AUC Percentile", np.percentile(scores, v))
    
#     auc_percentile = np.percentile(scores, v)
#     print(k, ": AUC Percentile", auc_percentile)
    print('\n'+"************")


    

top 10 submission scores [0.81724 0.81706 0.81652 0.81629 0.81584 0.81566 0.81562 0.81515 0.81502
 0.81471]
bottom 10 submission scores [0.25707 0.25591 0.25576 0.25559 0.2553  0.25511 0.2549  0.25473 0.25365
 0.25259]

************
dnn__weighted_cross_entropy_with_logits_exp/main_plus_all/submissions.csv 0.50995
Ranked 39504 of 40849 submissions
My submission was worse than 96.71% of submissions
My submission was better than 3.29% of submissions
AUC Percentile 0.48915956888

************

************
xgboost_kaggle_higgs_exp/main_plus_all/submissions.csv 0.68446
Ranked 36229 of 40849 submissions
My submission was worse than 88.69% of submissions
My submission was better than 11.31% of submissions
AUC Percentile 0.498422350912

************

************
xgboost_exp.max_depth_12/main_plus_all/submissions.csv 0.50495
Ranked 39625 of 40849 submissions
My submission was worse than 97.01% of submissions
My submission was better than 2.99% of submissions
AUC Percentile 0.48869526944

*****

### Best Model Performance

* xgboost_kaggle_higgs_exp/main_plus_all/submissions.csv 
* Score 0.68446 (best competition model scored 0.81471)
    * Ranked 36229 of 40849 submissions
    * My submission was worse than 88.69% of submissions
    * My submission was better than 11.31% of submissions
    * AUC Percentile 0.498422350912